samples/nrf9160/gnss/src/main.c

```diff
@@ -200,6 +231,87 @@ static void agps_data_get(struct k_work *item)
 }
 #endif /* !CONFIG_GNSS_SAMPLE_ASSISTANCE_NONE */
 
+#if defined(CONFIG_GNSS_SAMPLE_MODE_TTFF_TEST)
+static void ttff_test_got_fix_work_fn(struct k_work *item)
+{
+	LOG_INF("Time to fix: %u", time_to_fix);
+	LOG_INF("Sleeping for %u seconds", CONFIG_GNSS_SAMPLE_MODE_TTFF_TEST_INTERVAL);
+}
+
+static void ttff_test_force_cold_start(void)
+{
+	int err;
+	uint32_t delete_mask;
+
+	LOG_INF("Deleting GNSS NV data");
+
+	/* Delete everything else except the TCXO offset. */
+	delete_mask = NRF_MODEM_GNSS_DELETE_EPHEMERIDES |
+		      NRF_MODEM_GNSS_DELETE_ALMANACS |
+		      NRF_MODEM_GNSS_DELETE_IONO_CORRECTION_DATA |
+		      NRF_MODEM_GNSS_DELETE_LAST_GOOD_FIX |
+		      NRF_MODEM_GNSS_DELETE_GPS_TOW |
+		      NRF_MODEM_GNSS_DELETE_GPS_WEEK |
+		      NRF_MODEM_GNSS_DELETE_UTC_DATA |
+		      NRF_MODEM_GNSS_DELETE_GPS_TOW_PRECISION;
+
+	/* With minimal assistance, we want to keep the factory almanac. */
+	if (IS_ENABLED(CONFIG_GNSS_SAMPLE_ASSISTANCE_MINIMAL)) {
+		delete_mask &= ~NRF_MODEM_GNSS_DELETE_ALMANACS;
+	}
+
+	err = nrf_modem_gnss_nv_data_delete(delete_mask);
+	if (err) {
+		LOG_ERR("Failed to delete GNSS NV data");
```

Comment:
Should we actually abort the test if used request cold start but it cannot be done?

Reply:
Yes, you are right. It's better to abort if delete fails, because it wouldn't be a cold start test anyway.

Reply:
Done.